Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -293,10 +293,11 @@ public async Task ValidateContestPassword(int id, bool official, string password
                 };
             }
 
-            var contest = await this.contestsData.GetByIdWithProblemsDetailsAndCategories(model.ContestId);
+            var contest =
+                await this.contestParticipantsCacheService.GetContestServiceModelForContest(participant.ContestId, user, model);
 
             var validationResult = this.contestParticipationValidationService.GetValidationResult((
-                contest,
+                contest.Map<Contest>(),
```

Comment:
Can we modify the validation service to use the serviceModel directly?

Reply:
Done.